mbed-util/CriticalSectionLock.h

```diff
@@ -47,8 +47,7 @@ class CriticalSectionLock {
 #ifdef TARGET_NORDIC
         sd_nvic_critical_region_enter(&_state);
 #else
-        _state = __get_PRIMASK();
-        __disable_irq();
+        _state = minar::platform::pushDisableIRQState();
```

Comment:
I need to abstract away critical section entry and exit. These should not depend on hardware or constructs like PRIMASK; because that's not the level of detail we should be talking here; and that's not available on POSIX.

I believe CriticalSectionLock should encapsulate pushDisableIRQState(). Should I move pushDisableIRQState() into core-util within CriticalSectionLock?